### DISCUSSION TITLE
Does AggressivelyCache work when loading multiple documents lazily

### DISCUSSION THREAD

**Discussion by andrewdavey** [ravendb/ravendb#12438] [Q&A] [answered]
Given something like:

```csharp
using (session.Advanced.DocumentStore.AggressivelyCache())
{
    session.Advanced.Lazily
        .LoadAsync<Example>("Examples/1");

    session.Advanced.Lazily
        .LoadAsync<Example>("Examples/2");

    await session.Advanced.Eagerly.ExecuteAllPendingLazyOperationsAsync();
}
```

Should the resulting multi-get from the database be avoided when the documents are already cached in the document store?

Watching the traffic in Studio I see the multi-get request every time.

My web app needs to load five documents that are used by most requests it handles. The docs don't change often, so I think caching them would be good.

But if the multi-get isn't aggressively cached, then maybe it's better to do five regular `session.LoadAsync` calls instead?

**Comment by ayende** [comment, marked as answer]
What build are you running ? We had a bug where we wouldn't optimize this properly.
In general, if you run this twice on different sessions, this is expected to work and no hit the server.

The relevant code for that is here:
https://github.com/ravendb/ravendb/blob/ea47045927eaf2916bfad2f4cbd9f2e19215538e/src/Raven.Client/Documents/Commands/MultiGet/MultiGetCommand.cs#L40

This is about 3 months old, so you may have an older client version?

**Reply by andrewdavey** [reply]
Ah yes, I was on 5.0.8. Upgraded to 5.2 and it's working perfectly now. Thanks!

**Comment by aviv86** [comment]
tested this on latest 5.1 release and on 5.2, 
works as expected - documents are served from cache
